frontend/src/main.tsx

```diff
@@ -1,22 +1,33 @@
-import { createRouter, RouterProvider } from "@tanstack/react-router";
+import {StrictMode} from 'react';
 import ReactDOM from 'react-dom/client';
-import { routeTree } from "./routeTree.gen";
-import { StrictMode } from "react";
+import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
+import {createRouter, RouterProvider} from '@tanstack/react-router';
 
-const router = createRouter({ routeTree })
+import {routeTree} from './routeTree.gen';
+
+const queryClient = new QueryClient();
+const router = createRouter({
+  routeTree,
+  context: {
+    queryClient,
+  },
+  Wrap: ({children}) => {
+    return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
+  },
```

Comment:
Did not know this field existed, that is sooo nice for wrapping global providers with.